src/dune/file_tree.ml

```diff
@@ -368,15 +380,30 @@ end = struct
             "Note: You can use \"dune upgrade\" to convert your project to \
              dune."
         ]
-    else if not (String.Set.mem files Dune_file.fname) then
-      None
-    else (
-      ignore
-        ( Dune_project.ensure_project_file_exists project
-          : Dune_project.created_or_already_exist );
-      let file = Path.Source.relative path Dune_file.fname in
-      Some (Dune_file.load file ~project)
-    )
+    else
+      let file_exists = String.Set.mem files Dune_file.fname in
+      let parent =
+        let open Option.O in
+        let* parent = Path.Source.parent path in
+        let* parent = find_dir parent in
+        match parent.contents.dune_file with
+        | None
+        | Some (Ocaml_script _) ->
+          None (* wrong but easy *)
```

Comment:
@diml This means that stuff like this is broken:

```
$ cat dune
(chdir foo
  (chdir bar
   ...))
$ cat foo/dune
  (* -*- tuareg -*- *)
  ..
$ cat foo/bar/dune
  ;; random dune file
```

It can in theory be fixed, but it seems to me that it's better to just phase out ocaml scripts.

Reply:
Oh, I see now why it's a bit complicated. I guess the fix could be to produce an error if an OCaml script *inherits* any chdirs from its parent.